### PR TITLE
[RayService][Refactor] Change the ServeConfigs to nested map

### PR DESCRIPTION
**Pull request by MortalHappiness**
## Why are these changes needed?

See the description in the corresponding issue for details.

## Related issue number

Closes: ray-project/kuberay#2550

## Checks

- [x] I've made sure the tests are passing.
- Testing Strategy
   - [x] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by kevin85421** [comment]
cc @rueian would you mind reviewing the PR?

File: ray-operator/controllers/ray/rayservice_controller.go [hunk 1]
```diff
@@ -960,12 +956,27 @@ func (r *RayServiceReconciler) getAndCheckServeStatus(ctx context.Context, dashb
 	return isReady, nil
 }
 
-func (r *RayServiceReconciler) generateConfigKey(rayServiceInstance *rayv1.RayService, clusterName string) string {
-	return r.generateConfigKeyPrefix(rayServiceInstance) + clusterName
+func (r *RayServiceReconciler) getServeConfigFromCache(rayServiceInstance *rayv1.RayService, clusterName string) string {
+	cacheKey := rayServiceInstance.Namespace + "/" + rayServiceInstance.Name
+	serveConfigs, exist := r.ServeConfigs.Get(cacheKey)
+	if !exist {
+		return ""
```
Comment (rueian) [inline ray-operator/controllers/ray/rayservice_controller.go:963]: Is it possible that the config is indeed an empty string?

Comment (MortalHappiness) [reply ray-operator/controllers/ray/rayservice_controller.go:963]: Fixed in https://github.com/ray-project/kuberay/pull/2591/commits/040d4755b0ba54b48130c35456b72a5520ccc4e0. Only cache the serveConfig when it is not empty.

File: ray-operator/controllers/ray/rayservice_controller.go [hunk 2]
```diff
@@ -57,7 +57,7 @@ type RayServiceReconciler struct {
 	Recorder record.EventRecorder
 	// Currently, the Ray dashboard doesn't cache the Serve deployment config.
 	// To avoid reapplying the same config repeatedly, cache the config in this map.
-	ServeConfigs                 cmap.ConcurrentMap[string, string]
+	ServeConfigs                 cmap.ConcurrentMap[string, cmap.ConcurrentMap[string, string]]
```
Comment (kevin85421) [inline ray-operator/controllers/ray/rayservice_controller.go:60]: Can you add a comment to explain the definitions of key and value?

**Review by kevin85421 — APPROVED** [review]

File: ray-operator/controllers/ray/rayservice_controller.go (hunk 2 above)
Comment (MortalHappiness) [reply ray-operator/controllers/ray/rayservice_controller.go:62]: Added in https://github.com/ray-project/kuberay/pull/2591/commits/ca8482b26c48048a01bbcae4caf61325fa0cdf67

File: ray-operator/controllers/ray/rayservice_controller.go [hunk 3]
```diff
@@ -57,7 +57,9 @@ type RayServiceReconciler struct {
 	Recorder record.EventRecorder
 	// Currently, the Ray dashboard doesn't cache the Serve deployment config.
```
Comment (kevin85421) [inline ray-operator/controllers/ray/rayservice_controller.go:58]:
```suggestion
	// Currently, the Ray dashboard doesn't cache the Serve application config.
```

File: ray-operator/controllers/ray/rayservice_controller.go [hunk 4]
```diff
@@ -57,7 +57,9 @@ type RayServiceReconciler struct {
...
 	// To avoid reapplying the same config repeatedly, cache the config in this map.
-	ServeConfigs                 cmap.ConcurrentMap[string, string]
+	// Stores map of cacheKey to map of RayCluster name to Serve deployment config,
```
Comment (kevin85421) [inline ray-operator/controllers/ray/rayservice_controller.go:60]:
```suggestion
	// Stores map of cacheKey to map of RayCluster name to Serve application config,
```

File: ray-operator/controllers/ray/rayservice_controller.go [hunk 5]
```diff
@@ -529,21 +531,21 @@ func (r *RayServiceReconciler) getRayClusterByNamespacedName(ctx context.Context
 // cleanUpServeConfigCache cleans up the unused serve deployments config in the cached map.
```
Comment (kevin85421) [inline ray-operator/controllers/ray/rayservice_controller.go:531]:
```suggestion
// cleanUpServeConfigCache cleans up the unused serve application configs in the cached map.
```

**Review by kevin85421 — APPROVED** [review]
Nit: It should be "serve application configs" instead of "serve deployment configs."

**Pull request merged**